### PR TITLE
API/DEPR: replace kwarg "pat" with "sep" in str.[r]partition

### LINKED ISSUE #22676: DOC/DEPR: sep vs pat in series.str.partition
All the way back to the v0.16 docs, `Series.str.partition` (https://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.str.partition.html) has the signature ```Series.str.partition(pat=' ', expand=True)```, but the docs only talk about `sep` instead of `pat`.

I think the kwarg should be renamed to `sep`, which makes more sense for partitioning something than `pat`, which sounds like it's about a regex pattern.

Same thing applies to `rpartition`.

### PR DESCRIPTION
**Pull request by h-vetinari**
- [x] closes #22676
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/strings.py [hunk 1]
```diff
@@ -2410,8 +2410,11 @@ def rsplit(self, pat=None, n=-1, expand=False):
 
     Parameters
     ----------
-    pat : str, default whitespace
+    sep : str, default whitespace
```
Comment (WillAyd) [inline pandas/core/strings.py]: Let's still keep `pat` documented but with a `deprecated` directive

**Review by WillAyd — CHANGES_REQUESTED** [review]
Minor comment otherwise lgtm

File: pandas/core/strings.py [hunk 2]
```diff
@@ -2410,8 +2410,11 @@ def rsplit(self, pat=None, n=-1, expand=False):
...
         String to split on.
+    pat : str, default whitespace
+        .. deprecated:: 0.24.0
+           Use ``sep`` instead
```
Comment (WillAyd) [inline pandas/core/strings.py]: Thanks! Do we need a newline after this or does it render OK?

**Review by WillAyd — APPROVED** [review]

**Comment by h-vetinari** [comment]
Failure is a flaky hypothesis test.

**Comment by jreback** [comment]
looks fine, ex @WillAyd comments. @WillAyd pls add this to the list in #6581 as well. merge when ready.

**Comment by h-vetinari** [comment]
@jreback @WillAyd 
The feedback of @WillAyd is already incorporated.

**Comment by jreback** [comment]
thanks @h-vetinari

**Pull request merged**